source/common/http/async_client_impl.h

```diff
@@ -168,6 +169,13 @@ class AsyncStreamImpl : public AsyncClient::Stream,
     RouteEntryImpl route_entry_;
   };
 
+  struct NullSpan : public Tracing::Span {
+
+    // Tracing::Span
+    void setTag(const std::string&, const std::string&) override { NOT_IMPLEMENTED; }
+    void finishSpan() override { NOT_IMPLEMENTED; }
```

Comment:
I'd suggest moving NullSpan over into the Tracing namespace, and making these no-ops, rather than PANICs.

That way, a NullTracer and/or NullDriver can return a NullSpan instead of a nullptr, and this becomes somewhat safer to use.